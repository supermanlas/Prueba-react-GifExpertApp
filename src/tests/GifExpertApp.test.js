import React from 'react';
import { shallow } from 'enzyme'
import GifExpertApp from '../GifExpertApp'

// No se pueden hacer pruebas en el hook useState

describe('Prueba componente <GifExpertApp />', () => {

    test('Debe renderizar el componente', () => {

        const wrapper = shallow( <GifExpertApp />)
        expect( wrapper ).toMatchSnapshot();
        
    })
    
    test('Debe de mostrar una lista de categorÃ­as', () => {

        const categories = ['Haikyuu!','Detective Conan'];

        const wrapper = shallow( <GifExpertApp defaultCategories={ categories }/>)

        expect(wrapper).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
        
    })
    
    
})
