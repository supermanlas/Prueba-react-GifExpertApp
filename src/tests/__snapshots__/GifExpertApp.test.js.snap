// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba componente <GifExpertApp /> Debe de mostrar una lista de categor√≠as 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Haikyuu!"
      key="Haikyuu!"
    />
    <GifGrid
      category="Detective Conan"
      key="Detective Conan"
    />
  </ol>
</Fragment>
`;

exports[`Prueba componente <GifExpertApp /> Debe renderizar el componente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
